internal/config/db.go

```diff
@@ -70,6 +71,16 @@ type PostgresReadReplica struct {
 	Hostname string `yaml:"hostname" json:"hostname" env:"HOSTNAME"`
 }
 
+// PostgresPool defines configuration for connections pooling.
+type PostgresPool struct {
+	Name              string `yaml:"name" json:"name" env:"NAME" env-default:"pgx"`
+	MaxConns          int32  `yaml:"max_conns" json:"max_conns" env:"MAX_CONNS" env-default:"50"`
+	MinConns          int32  `yaml:"min_conns" json:"min_conns" env:"MIN_CONNS" env-default:"2"`
+	MaxConnLifetime   string `yaml:"max_conn_lifetime" json:"max_conn_lifetime" env:"MAX_CONN_LIFETIME" env-default:"1h"`
+	MaxConnIdleTime   string `yaml:"max_conn_idle_time" json:"max_conn_idle_time" env:"MAX_CONN_IDLE_TIME" env-default:"1h"`    //nolint:lll
+	HealthCheckPeriod string `yaml:"health_check_period" json:"health_check_period" env:"HEALTH_CHECK_PERIOD" env-default:"1m"` //nolint:lll
```

Comment:
Err, I linked to the wrong lib lol. Sorry, the library we're using for config unmarshalling also supports this:
https://github.com/ilyakaznacheev/cleanenv

We just need to change the type to `time.Duration`, and it'll just work as expected.

Reply:
ah! no idea why I thought Duration was supported in cleanenv without even looking :facepalm: ... maybe because of how `queryTimeout` was handled. fixed!